application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,412 @@
+            // that the format pattern defined in the config and the
+            // feed's actual format differ.
+            getZonedDateTime(firstRssFeedPubDate, feedConfig.dateFormatterPattern());
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to find a text channel from a given RSS feed configuration.
+     *
+     * @param jda the JDA instance
+     * @param feed the RSS feed configuration to search for a text channel
+     * @return an {@link Optional} containing the found text channel if it exists, otherwise empty
+     */
+    private Optional<TextChannel> getTextChannelFromFeed(JDA jda, RSSFeed feed) {
+        // Attempt to find the target channel, use the fallback otherwise
+        if (targetChannelPatterns.containsKey(feed)) {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> targetChannelPatterns.get(feed).test(channel.getName()))
+                .findFirst();
+        } else {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> fallbackChannelPattern.test(channel.getName()))
+                .findFirst();
+        }
+    }
+
+    /**
+     * Provides the {@link EmbedBuilder} from an RSS item used for sending RSS posts.
+     *
+     * @param item the RSS item to construct the embed message from
+     * @param feedConfig the configuration of the RSS feed
+     * @return the constructed {@link EmbedBuilder} containing information from the RSS item
+     */
+    private static EmbedBuilder constructEmbedMessage(Item item, RSSFeed feedConfig) {
+        final EmbedBuilder embedBuilder = new EmbedBuilder();
+        String title = item.getTitle().orElse("No title");
+        String titleLink = item.getLink().orElse("");
+        Optional<String> rawDescription = item.getDescription();
+
+        // Set the item's timestamp to the embed if found
+        item.getPubDate()
+            .ifPresent(date -> embedBuilder
+                .setTimestamp(getZonedDateTime(date, feedConfig.dateFormatterPattern())));
+
+        embedBuilder.setTitle(title, titleLink);
+        embedBuilder.setAuthor(item.getChannel().getLink());
+
+        // Process embed's description if a raw description was provided
+        if (rawDescription.isPresent() && !rawDescription.get().isEmpty()) {
+            Document fullDescription =
+                    Jsoup.parse(StringEscapeUtils.unescapeHtml4(rawDescription.get()));
+            String finalDescription = fullDescription.body()
+                .select("*")
+                .stream()
+                .map(Element::text)
+                .collect(Collectors.joining(". "));
+
+            embedBuilder.setDescription(StringUtils.abbreviate(finalDescription, MAX_CONTENTS));
+            return embedBuilder;
+        }
+
+        // Fill the description with a placeholder if the description was empty
```

Comment:
Put an else here, because the flow would make more sense, and delete this comment.

Reply:
Corrected!